Projects/ChangeTier/Sources/Service/Protocols/ChangeTierClient.swift

```diff
@@ -0,0 +1,3 @@
+public protocol ChangeTierClient {
+    func getTier(contractId: String, tierSource: ChangeTierSource) async throws -> ChangeTierIntentModel
```

Comment:
this function can throw. I think we should add specific errors for this (that we should also implement in other places) but this is a good start
```
public protocol ChangeTierClient {
    func getTier(contractId: String, tierSource: ChangeTierSource) async throws(ChangeTierError) -> ChangeTierIntentModel
}
```

```
public enum ChangeTierError: Error {
    case emptyList
    case somethingWentWrong
    case networkError
}
```
```
extension ChangeTierError: LocalizedError {
    var errorDescription: String? {
      ....
        }
    }
}
```